### ISSUE TITLE
Schema customisation: defining fields for the 'Query' root type

### ISSUE THREAD

**Issue by MorrisonCole** [gatsbyjs/gatsby#17495]
# Summary

I can't determine how to define types for the `Query` type Gatsby generates. I want to do this because, by default, Gatsby generates all fields as nullable, making working with Typescript very verbose!

## Relevant information

Using the blog starter project, I have the following query:
```
query {
  site {
    siteMetadata {
      title
      description
      author
    }
  }
}
```

I have defined custom non-nullable types for some data I know will always be present, e.g.,:

`gatsby-node.js`: 
```javascript
exports.createSchemaCustomization = ({ actions }) => {
  const { createTypes } = actions
  const typeDefs = `
    type Site implements Node {
      siteMetadata: SiteSiteMetadata!
    }
    
    type SiteSiteMetadata {
      title: String!
      description: String!
      author: String!
    }
  `
  createTypes(typeDefs)
}
```

I want to define `site` as non-nullable too, but when I add for example:

```javascript
type Query {
  site: Site!
}
```

It appears to be overwritten.

**TL;DR:** Is it possible to define custom fields for the root `Query` type?

**Comment by MorrisonCole** [comment]
Not stale! 👍 

Still wondering how to do this :)

**Comment by stefanprobst** [comment]
you *should* be able to use `createResolvers` to add custom root query fields, see https://www.gatsbyjs.org/docs/schema-customization/#custom-query-fields

**Comment by vladar** [comment]
Thank you for opening this, @MorrisonCole

We're marking this issue as answered and closing it for now but please feel free to reopen this and comment if you would like to continue this discussion. We hope we managed to help and thank you for using Gatsby! 💜
